lib/validate.js

```diff
@@ -1,82 +1,75 @@
-var merge = require('mout/object/merge')
+var jquery = require('./styles/jquery.js')
+var angular = require('./styles/angular.js')
+
+var limits = require('./utils/limits')
+var tickets = require('./utils/tickets')
+
+var defaults = require('./defaults.js')
+
 var semver = require('semver')
-var subjectExceptions = /^(fixup|squash)!|^\[[^\]]+\]:/
-var defaults = {
-  component: true,
-  components: [],
-  markerPattern: '^(clos|fix|resolv)(e[sd]|ing)',
-  actionPattern: '^([Cc]los|[Ff]ix|[Rr]esolv)(e[sd]|ing)\\s+[^\\s\\d]+(\\s|$)',
-  ticketPattern: '^(Closes|Fixes) (.*#|gh-|[A-Z]{2,}-)[0-9]+',
-  limits: {
-    subject: 72,
-    other: 80
-  }
-}
+var merge = require('mout/object/merge')
 
 module.exports = function (message, options) {
-  options = merge(defaults, options)
+  if (message === undefined) {
+    return ['Commit message is undefined, abort with error']
+  } else if (
+    !(typeof message === 'string' || message instanceof String)
+  ) {
+    return ['Commit message is not a string, abort with error']
+  } else if (message.length === 0) {
+    return ['Commit message is empty, abort with error']
+  }
+
+  if (options === undefined || options.style === undefined) {
+    options = merge(defaults.jquery, options)
+  } else if (options.style === 'jquery') {
+    options = merge(defaults.jquery, options)
+  } else if (options.style === 'angular') {
+    options = merge(defaults.angular, options)
+  } else {
+    return [
+      'Style ' + options.style + ' is not recognised\n' +
+      'Did you mistype it in package.json?'
+    ]
+  }
+
+  var lines = message.split('\n')
+
+  if (semver.valid(lines[0])) {
+    return []
+  }
+
+  if (lines[0].match(RegExp(options.wip))) {
```

Comment:
wip in particular, all the exceptions in general.

Since they use wip, seems fine to keep.

Still, in general, do we need options for these? I rather hardcode these exceptions and add options only when there a specific need.

A good way to see if we're adding too many options: Update the README with all the new options.